pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -532,6 +532,11 @@ def format_checkpoint_name(self, metrics: Dict[str, _METRIC], ver: Optional[int]
             >>> os.path.basename(ckpt.format_checkpoint_name(dict(epoch=2, val_loss=0.123456)))
             'epoch=2-val_loss=0.12.ckpt'
             >>> ckpt = ModelCheckpoint(dirpath=tmpdir,
+            ... filename='{epoch}-{val_loss:.2f}-{monitor:.2f}',
+            ... monitor='val_loss')
+            >>> os.path.basename(ckpt.format_checkpoint_name(dict(epoch=2, val_loss=0.123456)))
+            'epoch=2-val_loss=0.12-monitor=0.12.ckpt'
```

Comment:
IMO, I would prefer filename='{epoch}-{val_loss:.2f}-{monitor:.2f}' to be rendered 'epoch=2-val_loss=0.12-val_loss=0.12.ckpt' where monitor is being converted to val_loss.

Reply:
So users could just do filename='{epoch}-{monitor:.2f}' and easily change monitor metric name without changing the filename.

Reply:
Hello, @tchaton. I was thinking about it as well, but would it result in inconsistent logic with other metrics? The name for all other metrics will simply be their template strings. I am not sure, but I am glad to update this PR accordingly if you think the auto conversion would be better.

Reply:
Hi @tchaton, I reconsidered the problem and now agree with you. Keeping the keyword 'monitor' leads to the possibility of confusion between experiments with different monitor metrics. I just updated the PR.